### ISSUE TITLE
undo not working properly sometimes

### ISSUE THREAD

**Issue by davidperrenoud** [fritzing/fritzing-app#1462]
_From [irasc...@gmail.com](https://code.google.com/u/104729248032245122687/) on May 27, 2011 00:31:41_

Delete and undo-delete the orange wire connecting the piezo in bb view.  The piezo is no longer connected in schematic view.  

What we know about this file is that there was a whole set of deletions, followed by a bunch of undos.  The earlier version of the file (uploaded in issue #1460 ) doesn't have this problem

**Attachment:** [CosmicRayGun (2).fz](http://code.google.com/p/fritzing/issues/detail?id=1462)

_Original issue: http://code.google.com/p/fritzing/issues/detail?id=1462_

**Comment by failiz** [comment]
I could not replicate this in 0.9.4 and the file is not available.

**Comment by KjellMorgenstern** [comment]
I recently stumbled upon some QUndoCommands that looked broken:

```
+++ b/src/autoroute/autorouter.cpp
@@ -277,9 +277,9 @@ void Autorouter::addUndoConnection(bool connect, ConnectorItem * connectorItem,
                                ViewLayer::specFromID(toConnectorItem->attachedToViewLayerID()),
                                connect, parentCommand);
                ccc->setUpdateConnections(false);
        }
 }
```

Undo commands are created, more precise, subcomands. But I could not yet find where those commands are put into the m_commands list. It looks like ccc is just leaked. However, I can't say I fully understood this code yet, maybe it is not even called.

**Comment by failiz** [comment]
I found weird behaviours while doing undos, but never found a way to replicate them.

**Comment by failiz** [comment]
Remove imported label

**Comment by KjellMorgenstern** [comment]
The original issue described can not be reproduced.
Other issues seem to unspecific. We have solved some issues about undo, for example #3603 and #3720
